src/plugins/pcr/src/InSilicoPcrTask.h

```diff
@@ -106,6 +106,9 @@ class InSilicoPcrTask : public Task {
     void updateSequenceByPrimer(const PrimerBind &primer, QByteArray &productSequence) const;
 
 private:
+    //it takes ~6 minutes to process matrix 50x50 results with default algorithm settings in release build
```

Comment:
Why 6 minutes are good, but 1 hour is not? If it means "compute  fast and do not make users to wait" this limit should be much lower: 10-30 seconds

Is it possible to ask about this value (how long a user can wait)  in the related configuration dialog, so a user will see this value and will be able to adjust it?

Reply:
I just choose this value because with default values for this task and 3bp primers we wait 6 minutes and get result table with 706 results. As i think its more than enough for user to choose desired primer from results (or even decide what search parameters should be altered).
Anyway if you don't like my value feel free to suggest your.

Reply:
I would suggest adding an UI input: "Limit search time: N minutes".
This way users will be aware about this limit and will know what to expect.

The default value for the parameter can be set from the 'UGENE_MAX_RESULTS_FOR_PRIMERS_PER_STRAND' env variable.

Reply:
BTW, question: is it OK to show a table with 700 results to user? A set of this size looks unmanageable. What do you think?

Reply:
Ok, here is some hints for the final solution
https://www.google.com/search?q=pcr+product+length (I would suggest set a limit to 10k)
https://www.google.com/search?q=primer+length (my suggestion for the relaxed limit is 15-100)

With this number we do not need to solve/approve your current solution at all.
I think you can close this PR - this solution won't go. Lets start from a clean and correct one.